softioc/builder.py

```diff
@@ -181,6 +185,14 @@ def WaveformOut(name, *value, **fields):
 
 _DatabaseWritten = False
 
+def AddDatabaseLine(line, macros={}):
```

Comment:
This lets you add arbitrary lines to the database file. Is it named correctly, and could we do the macro replacement in a more efficient way?

Reply:
As I've commented above, I think I hate this API.  If you're going to build a database incrementally, you need to do it in a structured way.  Maybe parse the .db file and use the iocbuilder to rebuild it?  Not nice, I agree...